### PR TITLE
feat: [SSCA-1164]: Add bulk listing of tickets to the ticket service client

### PR DESCRIPTION
**Pull request by humanshu2002**
## Describe your changes
Added bulk listing of tickets to the ticket service client

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57094)

### CODE REVIEW

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger ut0

**Comment by humanshu2002** [comment]
trigger uts

**Review by mankrit-singh — DISMISSED** [review]

**Comment by humanshu2002** [comment]
trigger uts

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]
> Reviewed 10 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57094)* status: 1 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java` line 249 at r1](https://reviewable.io/reviews/harness/harness-core/57094#-Nme40Fd-Zjj26nr6kNx:-Nme40Fd-Zjj26nr6kNy:bg0cl2u) ([raw file](https://github.com/harness/harness-core/blob/5c647ea9d43a6bc716e62c75a50eaee7406631ff/ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java#L249)):*
> ```Java
>           String.format("Remediation Tracker: %s is already closed.", remediationTrackerId));
>     }
>     ArtifactInfo artifactInfo = remediationTracker.getArtifactInfos().get(body.getArtifactId());
> ```

only if body has artifactId we need to do this. otherwise ticket needs to be created for overall



_Code quote:_
```Java
 ArtifactInfo artifactInfo = remediationTracker.getArtifactInfos().get(body.getArtifactId());
```
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java` line 265 at r1](https://reviewable.io/reviews/harness/harness-core/57094#-Nme4CYT1gfadGsRnUzI:-Nme4CYT1gfadGsRnUzJ:b9zudy8) ([raw file](https://github.com/harness/harness-core/blob/5c647ea9d43a6bc716e62c75a50eaee7406631ff/ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java#L265)):*
> ```Java
>     String ticketId = ticketResponseDto.getExternalId();
> 
>     remediationTracker.setTicketId(ticketId);
> ```

you are always adding at top level for tracker. If artifactId is present add at artifact level otherwise at top level



_Code quote:_
```Java
    String ticketId = ticketResponseDto.getExternalId();

    remediationTracker.setTicketId(ticketId);
```
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/ticket/TicketServiceRestClient.java` line 21 at r1](https://reviewable.io/reviews/harness/harness-core/57094#-Nme4T9h2XiNemWSI7Gf:-Nme4T9h2XiNemWSI7Gg:b5ndk6s) ([raw file](https://github.com/harness/harness-core/blob/5c647ea9d43a6bc716e62c75a50eaee7406631ff/ssca-manager/service/src/main/java/io/harness/ssca/ticket/TicketServiceRestClient.java#L21)):*
> ```Java
> 
> public interface TicketServiceRestClient {
>   @POST("/tickets")
> ```

as discussed we would need to change default config for localhost and add /api



_Code quote:_
```Java
  @POST("/tickets")
```

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger smartchecks

**Comment by humanshu2002** [comment]
trigger pmd

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 1]
```diff
@@ -246,22 +246,30 @@ accountId, orgId, projectId, new ObjectId(remediationTrackerId))
       throw new InvalidArgumentsException(
           String.format("Remediation Tracker: %s is already closed.", remediationTrackerId));
     }
+
     ArtifactInfo artifactInfo = remediationTracker.getArtifactInfos().get(body.getArtifactId());
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:250]: this should be inside the if block

Comment (humanshu2002) [reply ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:250]: @Varsha-lalwani I am using the artifactInfo down as well , thats why kept it outside .

**Comment by humanshu2002** [comment]
trigger gitleaks

**Comment by humanshu2002** [comment]
trigger gitleaks

**Review by Varsha-lalwani — APPROVED** [review]

**Comment by Varsha-lalwani** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger ut9,ut7,ut3,ut2,ut1

**Review by mankrit-singh — APPROVED** [review]

**Pull request merged**